assignments/closure.js

```diff
@@ -1,21 +1,52 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
+var name = "Dora";
+function sayHello() {
+  return "Hey, my name is " + name + "!";
+}
 
+console.log(sayHello());
 
 /* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
 
-
 // ==== Challenge 2: Create a counter function ====
-const counter = () => {
-  // Return a function that when invoked increments and returns a counter variable.
-};
+// Return a function that when invoked increments and returns a counter variable.
 // Example usage: const newCounter = counter();
 // newCounter(); // 1
 // newCounter(); // 2
 
+const counter = () => {
+  let count = 0;
+  return function() {
+    count = count + 1;
+    return count;
+  };
+};
+
+const newCounter = counter();
+
```

Comment:
This is a closure, and an excellent example of why it is useful. The value of `count` is safe in the sense that it is only accessible using the function we can provide it. No other process can mutate its value out from under us.